src/test/scala/cromwell/util/SampleWdl.scala

```diff
@@ -976,4 +976,35 @@ object SampleWdl {
 
     override val rawInputs =  Map.empty[String, String]
   }
+
```

Comment:
I'd like to see a test along the lines of:
output {
 String s = array[ read_int("index.txt") ]
}

Also, I think the JES backend will not like this at all - in particular, preevaluateExpressionForFilenames might need to be updated.